src/mqt/bench/benchmark_generator.py

```diff
@@ -3,429 +3,382 @@
+        try:
+            qc = shor.create_circuit(instances[choice][0], instances[choice][1])
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(self.error_msg_outer_loop, "create_shor_benchmarks: ", choice, e)
+            raise e from None
+
+    def create_hhl_qc(self, index: int):
+        # index is not the number of qubits in this case
+        try:
+            # Creating the circuit on Algorithmic Description level
+            qc = hhl.create_circuit(index)
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(self.error_msg_outer_loop, "create_hhl_benchmarks", index, e)
+            raise e from None
+
+    def create_routing_qc(self, nodes: int):
+        try:
+            # Creating the circuit on Algorithmic Description level
+            qc = routing.create_circuit(nodes, 2)
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(self.error_msg_outer_loop, "create_routing_benchmarks", nodes, e)
+            raise e from None
+
+    def create_tsp_qc(self, nodes: int):
+        try:
+            # Creating the circuit on Algorithmic Description level
+            qc = tsp.create_circuit(nodes)
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(self.error_msg_outer_loop, "create_tsp_benchmarks", nodes, e)
+            raise e from None
+
+    def create_groundstate_qc(self, choice: str):
+        molecule = utils.get_molecule(choice)
+
+        try:
+            qc = groundstate.create_circuit(molecule)
+            qc.name = qc.name + "_" + choice
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(self.error_msg_outer_loop, "create_groundstate_benchmarks", choice, e)
+            raise e from None
+
+    def create_pricingcall_qc(self, num_uncertainty: int):
+        # num_options is not the number of qubits in this case
+        try:
+            # Creating the circuit on Algorithmic Description level
+            qc = pricingcall.create_circuit(num_uncertainty)
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(
+                self.error_msg_outer_loop,
+                "create_pricingcall_benchmarks",
+                num_uncertainty,
+                e,
+            )
+            raise e from None
+
+    def create_pricingput_qc(self, num_uncertainty: int):
+        # num_uncertainty is not the number of qubits in this case
+        try:
+            # Creating the circuit on Algorithmic Description level
+            qc = pricingput.create_circuit(num_uncertainty)
+            return qc, qc.num_qubits, False
+
+        except Exception as e:
+            print(
+                self.error_msg_outer_loop,
+                "create_pricingput_benchmarks",
+                num_uncertainty,
+                e,
+            )
+            raise e from None
```

Comment:
TBH, still not a really big fan of all of these functions that essentially do the same thing.
Can't these functions either be refactored to move some of the computation to the individual modules (like setting the name of a circuit, etc. or the exception handling which seems soooo repetitive here) or can these functions potentially be removed as a whole.
All of them return the same thing (`qc, qc.num_qubits, False` and do the same exception handling)